astropy/utils/data.py

```diff
@@ -1056,7 +1062,8 @@ def download_file(remote_url, cache=False, show_progress=True, timeout=None,
         result will be stored under the original URL. The original URL
         will *not* be tried unless it is in this list; this is to prevent
         long waits for a primary server that is known to be inaccessible
-        at the moment.
+        at the moment. If an empty list is passed, then ``download_file``
+        will not attempt to connect to the Internet.
```

Comment:
What do you mean? I thought if `sources=[]`, it would still attempt to connect to Internet via `remote_url`, right?

Please bear with me here. I am just channelling what a user might find confusing. Most users would use this without fully understanding all the internals.

Reply:
If you don't pass `sources` at all, or equivalently pass `sources=None`, you get the default of using `sources=[remote_url]`. If you pass an empty list, explicitly, that means "nowhere for you to get this"; I use that in `export_cache` to run `download_file` with a guarantee of never waiting for the Internet.

Reply:
Oh, gotcha! I missed that subtlety before.